File: src/main/java/com/dmforu/cafeteria/CafeteriaService.java
package com.dmforu.cafeteria;

import com.dmforu.crawling.parser.Parser;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class CafeteriaService {

    private final Parser<Diet> cafeteriaParser;

    public List<Diet> getData() {
        return cafeteriaParser.parse();
    }
}

File: src/main/java/com/dmforu/cafeteria/CafeteriaService.kt
package com.dmforu.cafeteria

import com.dmforu.crawling.parser.Parser
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class CafeteriaService {
    private val cafeteriaParser: Parser<Diet>? = null

    val data: List<Diet>
        get() = cafeteriaParser!!.parse()
}
